src/main/java/uk/gov/pay/products/service/PaymentDeleter.java

```diff
@@ -0,0 +1,44 @@
+package uk.gov.pay.products.service;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.gov.pay.products.config.ExpungeHistoricalDataConfig;
+import uk.gov.pay.products.persistence.dao.PaymentDao;
+import uk.gov.pay.products.persistence.entity.PaymentEntity;
+
+import javax.inject.Inject;
+import java.time.Clock;
+import java.time.temporal.ChronoUnit;
+import java.util.stream.Collectors;
+
+import static java.lang.String.format;
+import static java.time.ZoneOffset.UTC;
+
+public class PaymentDeleter {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(PaymentDeleter.class);
+    
+    private ExpungeHistoricalDataConfig expungeHistoricalDataConfig;
+    private PaymentDao paymentDao;
+    private Clock clock;
+
+    @Inject
+    public PaymentDeleter(ExpungeHistoricalDataConfig expungeHistoricalDataConfig, PaymentDao paymentDao, Clock clock) {
+        this.expungeHistoricalDataConfig = expungeHistoricalDataConfig;
+        this.paymentDao = paymentDao;
+        this.clock = clock;
+    }
+
+    public void deletePayments() {
+        if (!expungeHistoricalDataConfig.isExpungeHistoricalDataEnabled()) {
+            LOGGER.info("deletePayments was called but expunging of historical data is not enabled.");
+            return;
+        }
+        
+        var maxDate = clock.instant().minus(expungeHistoricalDataConfig.getExpungeDataOlderThanDays(), ChronoUnit.DAYS).atZone(UTC);
+        var forDeletion = paymentDao.getPaymentsForDeletion(maxDate, expungeHistoricalDataConfig.getNumberOfTransactionsToRedact())
```

Comment:
Is there any reason we are querying and deleting payments in separate calls?

Can it be one call `deletePayments(maxDate, expungeHistoricalDataConfig.getNumberOfTransactionsToDelete())`?

Reply:
Do you mean if we collapse both `paymentDao.getPaymentsForDeletion` and `paymentDao.deletePayments` into one line or merge both paymentDao methods into one?

Reply:
merge methods into one. One DB call to delete payments.

Reply:
I guess there's no reason to query and delete in separate calls; I was probably following [ledger's pattern of finding transactions first](https://github.com/alphagov/pay-ledger/blob/master/src/main/java/uk/gov/pay/ledger/transaction/dao/TransactionDao.java#L378).

Reply:
yeah no need to query payments in products db. We need it in ledger for logging and to delete related events.